File: src/client-app/src/models/requests/users/registerUserCommand.ts
export interface RegisterUserCommand {
    email: string
    username: string
    password: string
    currency: string
}

export class RegisterUserCommand implements RegisterUserCommand {
    constructor() {
        this.email = ''
        this.username = ''
        this.password = ''
        this.currency = 'USD'
    }
}

File: src/PennyPlanner.Application/Users/RegisterUser/RegisterUserCommand.cs
using CommonAbstractions.DB.Messaging;
using PennyPlanner.Domain.Users;

namespace PennyPlanner.Application.Users.RegisterUser
{
    public sealed record RegisterUserCommand(
        string Username,
        string Email,
        string Password,
        string Currency) : ICommand<User>
    {
    }
}
